src/middleware/queries.ts

```diff
@@ -141,10 +412,91 @@ export function eventByIndexedArgs(
       }
     }
   `;
+  const { filter, args } = createEventFilter(variables);
+  const queryV2 = gql`
+    query EventByIndexedArgsQuery($moduleId: String!, $eventId: String! ${args}) {
+      events(
+        filter: { moduleId: { equalTo: $moduleId }, eventId: { equalTo: $eventId } ${filter} }
+        first: 1
+      ) {
+        nodes {
+          blockId
+          eventIdx
+          extrinsicIdx
+          block {
+            datetime
+          }
+        }
+      }
+    }
+  `;
 
   return {
-    query,
-    variables,
+    v1: {
+      query,
+      variables,
+    },
+    v2: {
+      request: { query: queryV2, variables },
+      mapper: ({ events }) => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const { nodes } = events!;
+        const event = nodes[0];
+        if (!event) {
+          return {
+            eventByIndexedArgs: null,
+          };
+        }
+        return {
+          eventByIndexedArgs: mapEvent(event),
+        };
+      },
+    },
+  };
+}
+
+/**
+ *  @hidden
+ */
+function mapEvent({ blockId, eventIdx, extrinsicIdx, block }: EventV2): DeepPartial<EventV1> {
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { datetime } = block!;
+  return {
+    /* eslint-disable @typescript-eslint/naming-convention */
+    __typename: 'Event',
+    block_id: +blockId,
+    event_idx: eventIdx,
+    extrinsic_idx: extrinsicIdx,
+    block: { __typename: 'Block', datetime },
+    /* eslint-enable @typescript-eslint/naming-convention */
+  };
+}
+
+/**
+ *  @hidden
+ */
+function createEventFilter(variables: QueryEventByIndexedArgsArgs): {
+  filter: string;
+  args: string;
+} {
+  let filter = '';
+  let args = ' ';
+  const { eventArg0, eventArg1, eventArg2 } = variables;
+  if (eventArg0 !== undefined) {
+    args += ', $eventArg0: String';
```

Comment:
building up strings seems like a smell to me. Why doesn't Apollo handle this for us?